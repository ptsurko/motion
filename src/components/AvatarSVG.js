import React, { PureComponent } from 'react';
import Types from 'prop-types';
import { SVGAvatar } from '@satori-sdk/component-library';

class AvatarSVG extends PureComponent {
  shouldComponentUpdate(nextProps, nextState) {
    return false;
  }

  render() {
    const { avatar, color } = this.props;

    return (
      <SVGAvatar
        avatar={avatar}
        color={color}
        className="phone-avatar"
      />
    );
  }
}

AvatarSVG.propTypes = {
  avatar: Types.string.isRequired,
  color: Types.string.isRequired,
};

export default AvatarSVG;
