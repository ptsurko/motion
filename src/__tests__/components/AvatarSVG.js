import React from 'react';
import renderer from 'react-test-renderer';
import AvatarSVG from 'components/AvatarSVG';

it('renders AvatarSVG component correctly', () => {
  const tree = renderer.create(
    <AvatarSVG avatar="avatar1" color="#000000" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
