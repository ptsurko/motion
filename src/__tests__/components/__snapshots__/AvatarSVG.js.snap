// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders AvatarSVG component correctly 1`] = `
<div
  className="phone-avatar"
  onClick={[Function]}
>
  <svg
    dangerouslySetInnerHTML={
      Object {
        "__html": "
          <title>1</title>
          <g id=\\"Page-1\\" stroke=\\"none\\" stroke-width=\\"1\\" fill=\\"none\\" fill-rule=\\"evenodd\\">
              <g id=\\"1\\">
                  <g id=\\"Avatar_1\\">
                      <path d=\\"M50,99.009901 C22.3855556,99.009901 0,76.8470847 0,49.5049505 C0,22.1639164 22.3855556,0 50,0 C77.6144444,0 100,22.1639164 100,49.5049505 C100,76.8470847 77.6144444,99.009901 50,99.009901 Z\\" id=\\"bg\\" fill=\\"#000000\\"></path>
                      <g id=\\"Group\\" stroke-width=\\"1\\" transform=\\"translate(10.000000, 18.000000)\\">
                          <g id=\\"Avatar1\\">
                              <path d=\\"M17.0334247,6.86005479 L0.0153424658,38.4589589 L17.0334247,63.0036164 L61.8136986,63.0036164 L79.7260274,38.4589589 L61.8136986,6.86005479 L17.0334247,6.86005479 Z M17.3347945,44.6858082 L17.3347945,33.9406027 L27.9375342,22.5652603 L51.8038356,22.5652603 L62.4076712,33.9406027 L62.4076712,44.6858082 L51.8038356,53.8485479 L27.9375342,53.8485479 L17.3347945,44.6858082 Z\\" id=\\"Fill-1\\" fill=\\"#FFFFFF\\"></path>
                              <g id=\\"Group-13\\" transform=\\"translate(0.000000, 0.537534)\\">
                                  <path d=\\"M33.1438904,43.957589 C35.4726575,49.3921096 44.7899178,49.3921096 47.1197808,43.957589\\" id=\\"Stroke-2\\" stroke=\\"#FFFFFF\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\"></path>
                                  <path d=\\"M32.4604932,33.5156164 C32.4604932,35.3315068 30.9887123,36.8032877 29.1728219,36.8032877 C27.3569315,36.8032877 25.8851507,35.3315068 25.8851507,33.5156164 C25.8851507,31.699726 27.3569315,30.2279452 29.1728219,30.2279452 C30.9887123,30.2279452 32.4604932,31.699726 32.4604932,33.5156164\\" id=\\"Fill-4\\" fill=\\"#FFFFFF\\"></path>
                                  <path d=\\"M54.1653699,33.6076712 C54.1653699,35.4235616 52.693589,36.8953425 50.8776986,36.8953425 C49.0618082,36.8953425 47.5900274,35.4235616 47.5900274,33.6076712 C47.5900274,31.7928767 49.0618082,30.32 50.8776986,30.32 C52.693589,30.32 54.1653699,31.7928767 54.1653699,33.6076712\\" id=\\"Fill-6\\" fill=\\"#FFFFFF\\"></path>
                                  <path d=\\"M0.000109589041,37.9141918 L3.57380822,2.82268493 C3.57380822,2.82268493 3.28778082,1.82980822 4.76065753,1.17884932 C4.76065753,1.17884932 5.816,0.439123288 6.17654795,0.584876712 C7.29545205,1.04186301 6.8790137,1.89775342 6.86147945,2.54871233 C6.83846575,3.3750137 6.14586301,27.3037808 6.14586301,27.3037808 L0.000109589041,37.9141918 Z\\" id=\\"Fill-8\\" fill=\\"#FFFFFF\\"></path>
                                  <g id=\\"Group-12\\" transform=\\"translate(72.328767, 0.000000)\\">
                                      <mask id=\\"mask-2\\" fill=\\"white\\">
                                          <use xlink:href=\\"#path-1\\"></use>
                                      </mask>
                                      <g id=\\"Clip-11\\"></g>
                                      <path d=\\"M7.39736986,38.0051507 L3.82367123,2.91364384 C3.82367123,2.91364384 4.10969863,1.92076712 2.63682192,1.26980822 C2.63682192,1.26980822 1.58147945,0.530082192 1.22093151,0.675835616 C0.102027397,1.13282192 0.518465753,1.98980822 0.536,2.63967123 C0.559013699,3.46706849 1.25161644,27.3947397 1.25161644,27.3947397 L7.39736986,38.0051507 Z\\" id=\\"Fill-10\\" fill=\\"#FFFFFF\\" mask=\\"url(#mask-2)\\"></path>
                                  </g>
                              </g>
                          </g>
                      </g>
                  </g>
              </g>
          </g>
      ",
      }
    }
    viewBox="0 0 100 100"
    width={undefined}
  />
</div>
`;
